tests/table/tables_plugin.test.ts

```diff
@@ -0,0 +1,836 @@
+
+      deleteContent(model, ["A1:A5"]);
+      expect(getTable(model, "A1")).toBeFalsy();
+    });
+  });
+
+  describe("Table Zone Expansion", () => {
+    test("Table zone is expanded when creating a new cell just below the table", () => {
+      createTable(model, "A1:B3");
+      updateFilter(model, "A1", ["C"]);
+      setCellContent(model, "A4", "Something");
+      setCellContent(model, "B5", "Something Else");
+      expect(zoneToXc(model.getters.getTables(sheetId)[0].zone)).toEqual("A1:B5");
+      expect(getFilterValues(model)).toEqual([
+        { zone: "A1:A5", value: ["C"] },
+        { zone: "B1:B5", value: [] },
+      ]);
+    });
+
+    test("Table zone is expanded when creating cells right of the table", () => {
+      createTable(model, "B1:B3");
+      updateFilter(model, "B1", ["C"]);
+      setCellContent(model, "A1", "Something");
+      setCellContent(model, "C3", "Something Else");
+      expect(zoneToXc(model.getters.getTables(sheetId)[0].zone)).toEqual("B1:C3");
+      expect(getFilterValues(model)).toEqual([
+        { zone: "B1:B3", value: ["C"] },
+        { zone: "C1:C3", value: [] },
+      ]);
+    });
+
+    test("Table zone isn't expended at creation", () => {
+      setCellContent(model, "A4", "Something");
+      createTable(model, "A1:A3");
+      expect(zoneToXc(model.getters.getTables(sheetId)[0].zone)).toEqual("A1:A3");
+      expect(getFilterValues(model)).toEqual([{ zone: "A1:A3", value: [] }]);
+    });
+
+    test("Table zone isn't expanded when modifying existing cell", () => {
+      setCellContent(model, "A4", "Something");
+      createTable(model, "A1:A3");
+      setCellContent(model, "A4", "Something Else");
+      expect(zoneToXc(model.getters.getTables(sheetId)[0].zone)).toEqual("A1:A3");
+      expect(getFilterValues(model)).toEqual([{ zone: "A1:A3", value: [] }]);
+    });
+
+    test("Table zone isn't expanded when another cell below the table had a content", () => {
+      setCellContent(model, "B4", "Something");
+      createTable(model, "A1:B3");
+      setCellContent(model, "A4", "Something Else");
+      expect(zoneToXc(model.getters.getTables(sheetId)[0].zone)).toEqual("A1:B3");
+    });
+
+    test("Table zone is expanded when another cell below the table had a a style but no content", () => {
+      setStyle(model, "B4", { fillColor: "#000000" });
+      createTable(model, "A1:B3");
+      setCellContent(model, "A4", "Something");
+      expect(zoneToXc(model.getters.getTables(sheetId)[0].zone)).toEqual("A1:B4");
+    });
+
+    test("Table zone is expanded when a cell on the row below the table but not below the table had a content", () => {
+      setCellContent(model, "B4", "Something");
+      createTable(model, "A1:A3");
+      setCellContent(model, "A4", "Something Else");
+      expect(zoneToXc(model.getters.getTables(sheetId)[0].zone)).toEqual("A1:A4");
+    });
+
+    test("Table zone isn't expanded when there was another table table below it", () => {
```

Comment:
I would CTRL+H all `isn't` in the file to `is not` to avoid the single quote that breaks vscode jest extension